BE/db/seed.js

```diff
@@ -0,0 +1,425 @@
+    } catch (error) {
+      console.log("Error creating comments!");
+      throw error;
+    }
+  }
+  
+  async function rebuildDB() {
+    try {
+      client.connect();
+  
+      await dropTables();
+      await createTables();
+      await createInitialUsers();
+      await createInitialRecipes();
+      await createInitialReviews();
+      await createInitialComments();
+    } catch (error) {
+      console.log("Error during rebuildDB")
+      throw error;
+    }
+  }
+  
+  async function testDB() {
+    try {
+      console.log("Starting to test database...");
+  
+      console.log("Calling getAllUsers");
+      const users = await getAllUsers();
+      console.log("Result:", users);
+  
+      console.log("Calling updateUser on users[0]");
+      const updateUserResult = await updateUser(users[0].id, {
+        name: "Admin Test Updated Name"
+      });
+      console.log("Result:", updateUserResult);
+  
+      console.log("Calling getAllRecipes");
+      const recipes = await getAllRecipes();
+      console.log("Result:", recipes);
+  
+      console.log("Calling updateRecipe on recipes[0]");
+      const updateRecipeResult = await updateRecipe(recipes[0].id, {
+        title: "New Title",
+        procedure: ["Updated Content"]
+      });
+      console.log("Result:", updateRecipeResult);
+  
+      console.log("Calling updateRecipe on recipes[1], only updating tags");
+      const updateRecipeTagsResult = await updateRecipe(recipes[1].id, {
+        tags: ["pastries", "classic", "salads"]
+      });
+      console.log("Result:", updateRecipeTagsResult);
+
+      console.log("Calling getAllReviews");
+      const reviews = await getAllReviews();
+      console.log("Result:", reviews);
+  
+      console.log("Calling updateReview on reviews[0]");
+      const updateReviewResult = await updateReview(reviews[0].id, {
+        content: "Updated Content"
+      });
+      console.log("Result:", updateReviewResult);
+
+      console.log("Calling getAllComments");
+      const comments = await getAllComments();
+      console.log("Result:", comments);
+  
+      console.log("Calling updateComment on comments[0]");
+      const updateCommentResult = await updateComment(comments[0].id, {
+        content: "Updated Content"
+      });
+      console.log("Result:", updateCommentResult);
+  
+      console.log("Calling getUserById with 1");
+      const admin = await getUserById(1);
+      console.log("Result:", admin);
+  
+      console.log("Calling getAllTags");
+      const allTags = await getAllTags();
+      console.log("Result:", allTags);
+  
+      console.log("Calling getRecipesByTagName with salads");
+      const recipesWithGrandma = await getRecipesByTagName("salads");
+      console.log("Result:", recipesWithGrandma);
+  
+      console.log("Finished database tests!");
+    } catch (error) {
+      console.log("Error during testDB");
+      throw error;
+    }
+  }
+  
+  
+  rebuildDB()
```

Comment:
Might be worth refactoring this to use async await syntax.

Reply:
Lets go over this on tuesday